modules/users/server/models/user.server.model.js

```diff
@@ -121,20 +146,126 @@ UserSchema.methods.hashPassword = function(password) {
 /**
  * Create instance method for authenticating user
  */
-UserSchema.methods.authenticate = function(password) {
+UserSchema.methods.authenticate = function (password) {
 	return this.password === this.hashPassword(password);
 };
 
 /**
+ * Create instance method for authenticating user
+ */
+UserSchema.methods.setIdentity = function (provider, id, accessToken, refreshToken, providerData) {
+	var found = false;
+	var auth = {
+		provider: provider,
+		id: id
+	};
+	if (!this.identities) {
+		this.identities = [];
+	}
+	this.identities.forEach(function (identity) {
+		if (identity.provider === provider && identity.id === id) {
```

Comment:
Some strategies return ID as Number (yay!) - need to change to `identity.id === id`